src/core/lib/security/transport/security_connector.h

```diff
@@ -106,12 +98,12 @@ void grpc_security_connector_unref(grpc_security_connector *policy);
 #endif
 
 /* Check the peer. Callee takes ownership of the peer object.
-   The callback will include the resulting auth_context. */
+   Sets *auth_context and invokes on_peer_checked when done. */
 void grpc_security_connector_check_peer(grpc_exec_ctx *exec_ctx,
                                         grpc_security_connector *sc,
                                         tsi_peer peer,
-                                        grpc_security_peer_check_cb cb,
-                                        void *user_data);
+                                        grpc_auth_context **auth_context,
```

Comment:
The reason for doing this is that, now that we're executing the callback asynchronously via the exec_ctx, it needs to be a grpc_closure, which means that the argument is pre-bound by the caller -- there's no easy way to pass in a grpc_closure and have this function pass the grpc_auth_context directly to it when it's invoked.

The alternative would be to have the caller pass in a custom callback type and make it the responsibility of the check_peer() function to invoke its own callback via the exec_ctx, which would then directly invoke the caller-supplied custom callback type.  However, that seems fairly cumbersome, and it requires more memory allocation (which is more expensive), so I prefer the approach I've chosen here.

Also, this API is consistent with other uses within the C-core code base.  For example, look at grpc_tcp_client_connect() (https://github.com/grpc/grpc/blob/master/src/core/lib/iomgr/tcp_client.h#L52), which takes a grpc_endpoint** as an argument, which it fills in before calling the on_connect callback.